Install from URL doesn't work for multiply-nested GitLab projects

**Describe the bug**
Ajour turns a URL like `https://gitlab.com/thing-engineering/wowthing/wowthing-collector` into `https://gitlab.com/api/v4/projects/thing-engineering%2Fwowthing/releases`, when the project really is nested 3 levels deep (organization/subgroup/project).

**To Reproduce**
Steps to reproduce the behavior:
1. Click on `Install from URL`
2. Paste in `https://gitlab.com/thing-engineering/wowthing/wowthing-collector`
3. Click on `Install for Retail`
4. See error

**Expected behavior**
Install to proceed correctly.

**Software involved**
Please complete the following information:
 - OS: Windows 10
 - Ajour version: 1.2.0

**Log Output**
```
23:39:21.670 [ajour::gui::update][DEBUG] Interaction::ModeSelected(Install)
23:39:26.153 [ajour::gui::update][DEBUG] Interaction::InstallAddon(Retail, Source)
23:39:26.647 [ajour][ERROR] No release at https://gitlab.com/api/v4/projects/thing-engineering%2Fwowthing/releases
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -20,6 +20,10 @@ and `Removed`.
   - Polish (thanks KasprzykM)
 - Ability to select Screenshots when backing up.
 
+### Fixed
+
+- Fixed error importing Gitlab addons from repos under a nested subgroup
+
 ## [1.2.0] - 2021-05-18
 
 ### Added

--- a/crates/core/Cargo.toml
+++ b/crates/core/Cargo.toml
@@ -48,6 +48,7 @@ path-slash = "0.1.4"
 tar = "0.4.33"
 zstd = { version = "0.6.1", features = ["zstdmt"] }
 num_cpus = "1.13.0"
+urlencoding = "1.3.3"
 
 iced_native = { version = "0.4.0", optional = true }
 

--- a/crates/core/src/repository/backend/git.rs
+++ b/crates/core/src/repository/backend/git.rs
@@ -301,21 +301,20 @@ mod gitlab {
     #[async_trait]
     impl Backend for Gitlab {
         async fn get_metadata(&self) -> Result<RepositoryMetadata, RepositoryError> {
-            let mut path = self.url.path().split('/');
-            // Get rid of leading slash
-            path.next();
+            let mut path = self.url.path();
+            // Remove leading slash
+            path = path.trim_start_matches('/');
 
-            let author = path.next().ok_or(RepositoryError::GitMissingAuthor {
+            // Encode entire path for API
+            let encoded = urlencoding::encode(path);
+
+            // Get last path part as repo name
+            let path_parts = path.split('/');
+            let repo = path_parts.last().ok_or(RepositoryError::GitMissingRepo {
                 url: self.url.to_string(),
             })?;
-            let repo = path.next().ok_or(RepositoryError::GitMissingRepo {
-                url: self.url.to_string(),
-            })?;
 
-            let url = format!(
-                "https://gitlab.com/api/v4/projects/{}%2F{}/releases",
-                author, repo
-            );
+            let url = format!("https://gitlab.com/api/v4/projects/{}/releases", &encoded);
 
             let mut resp = request_async(&url, vec![], None).await?;
 
@@ -407,20 +406,16 @@ mod gitlab {
         ) -> Result<Option<String>, RepositoryError> {
             let tag_name = tag_name.ok_or(RepositoryError::GitChangelogTagName)?;
 
-            let mut path = self.url.path().split('/');
-            // Get rid of leading slash
-            path.next();
+            let mut path = self.url.path();
+            // Remove leading slash
+            path = path.trim_start_matches('/');
 
-            let author = path.next().ok_or(RepositoryError::GitMissingAuthor {
-                url: self.url.to_string(),
-            })?;
-            let repo = path.next().ok_or(RepositoryError::GitMissingRepo {
-                url: self.url.to_string(),
-            })?;
+            // Encode entire path for API
+            let encoded = urlencoding::encode(path);
 
             let url = format!(
-                "https://gitlab.com/api/v4/projects/{}%2F{}/releases/{}",
-                author, repo, tag_name
+                "https://gitlab.com/api/v4/projects/{}/releases/{}",
+                encoded, tag_name
             );
 
             let mut resp = request_async(&url, vec![], None).await?;

